src/autoscaler-common/counters_base.js

```diff
@@ -0,0 +1,448 @@
+    // them.
+    this.suppressErrors = false;
+  }
+
+  // eslint-disable-next-line require-jsdoc
+  verbose(message, ...args) {
+    logger.trace('otel: '+message, args);
+  }
+  // eslint-disable-next-line require-jsdoc
+  debug(message, ...args) {
+    logger.debug('otel: '+message, args);
+  }
+  // eslint-disable-next-line require-jsdoc
+  info(message, ...args) {
+    logger.info('otel: '+message, args);
+  }
+  // eslint-disable-next-line require-jsdoc
+  warn(message, ...args) {
+    logger.warn('otel: '+message, args);
+  }
+  // eslint-disable-next-line require-jsdoc
+  error(message, ...args) {
+    if ( ! this.suppressErrors ) {
+      logger.error('otel: '+message, args);
+    }
+  }
+};
+
+const DIAG_BUNYAN_LOGGER = new DiagToBunyanLogger();
+
+/**
+ * Number of errors reported by OpenTelemetry. Used by tryFlush() to detect
+ * if the flush suceeded or not.
+ */
+let openTelemetryErrorCount = 0;
+
+/**
+ * Global Error hander function for open Telemetry. Keeps a track of the
+ * number of errors reported.
+ *
+ * @param {Object} err
+ */
+function openTelemetryGlobalErrorHandler(err) {
+  openTelemetryErrorCount++;
+  // delegate to Otel's own error handler for stringification
+  OpenTelemetryCore.loggingErrorHandler()(err);
+}
+
+
+// Setup OpenTelemetry client libraries logging.
+OpenTelemetryApi.default.diag.setLogger(
+    DIAG_BUNYAN_LOGGER,
+    {
+      logLevel: OpenTelemetryApi.DiagLogLevel.INFO,
+      suppressOverrideMessage: true,
+    });
+OpenTelemetryCore.setGlobalErrorHandler(openTelemetryGlobalErrorHandler);
+
+
+/**
+ * Initialize the OpenTelemetry metrics, set up logging
+ *
+ * If called more than once, will wait for the first call to complete.
+ *
+ * @return {!Promise<void>}
+ */
+async function initMetrics() {
+  // check to see if someone else has started to init counters before
+  // so that this function only runs once.
+  if (pendingInit) {
+    return await pendingInit;
+  }
+
+  /** @type {?function():void} */
+  let resolvePendingInit = null;
+  /** @type {?function(?):void} */
+  let rejectPendingInit = null;
+  pendingInit = new Promise((res, rej) => {
+    resolvePendingInit = res;
+    rejectPendingInit = rej;
+  });
+
+
+  try {
+    logger.debug('initializing metrics');
+
+    if (process.env.KUBERNETES_SERVICE_HOST) {
+      // In K8s. We need to set the Pod Name to prevent duplicate
+      // timeseries errors.
+      if (process.env.K8S_POD_NAME) {
+        RESOURCE_ATTRIBUTES[
+            Semconv.K8S_POD_NAME] =
+            process.env.K8S_POD_NAME;
+      } else {
+        logger.warn('WARNING: running under Kubernetes, but K8S_POD_NAME ' +
+        'environment variable is not set. ' +
+        'This may lead to Send TimeSeries errors');
+      }
+    }
+
+    const gcpResources = new GcpDetectorSync().detect();
+    await gcpResources.waitForAsyncAttributes();
+    logger.debug('Got gcpResources attrs:1 %o', gcpResources.attributes);
```

Comment:
Nit: the formatting of the object in the debug message body takes up multiline real-estate in the top-level logging views in the web console -- do you think it's worth embedding this in a different field?

Reply:
Also, that was debug code I left in! Removing!